### DISCUSSION TITLE
Rails 8

### DISCUSSION THREAD

**Discussion by stephan-buckmaster** [AllYourBot/hostedgpt#519] [General]
Has anyone looked into upgrading the app to Rails 8? https://edgeguides.rubyonrails.org/8_0_release_notes.html says to go to 7.2 first (we're at 7.1.3)

**Comment by keithschacht** [comment]
Thanks for flagging this, I’ve been thinking about this as well. @jlvallelonga is in the process of getting Solid Queue upgraded to 1.0 which is a nice step. @jlvallelonga maybe you could handle the upgrade to rails 7.2 soon as well?

I took a look at the 7.2 upgrade guide and it should be straightforward: https://edgeguides.rubyonrails.org/7_2_release_notes.html

Most of it is configuration that the app already has and I wouldn’t want to revert their defaults (e.g. rubocop, ci). I’m not familiar with Brakeman so that’s probably worth trying their default config. Also, I don’t quite understand the difference between dev containers and docker. It seems like maybe it’s a wrapper around docker? If this is a new default it might be worth supporting.

Most likely the only material change will be adding the browser version guard feature.

**Reply by jlvallelonga** [reply]
ok I have a PR up here: https://github.com/AllYourBot/hostedgpt/pull/523
I'll work towards getting the 7.2 defaults in place and then remove the new_framework_defaults_7_2.rb file
